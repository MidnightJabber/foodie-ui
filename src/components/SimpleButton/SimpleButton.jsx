import React from 'react';
import PropTypes from 'prop-types';

import './SimpleButton.scss';

const SimpleButton = ({
  title,
  type,
  onBtnClick,
  btnDisabled,
}) => (
  <button
    type={type}
    className="simplebutton"
    disabled={btnDisabled}
    onClick={onBtnClick}
  >
    {title}
  </button>
);

SimpleButton.propTypes = {
  title: PropTypes.string,
  type: PropTypes.string,
  btnDisabled: PropTypes.bool,
  onBtnClick: PropTypes.func,
};

SimpleButton.defaultProps = {
  title: '',
  type: 'button',
  btnDisabled: false,
  onBtnClick: () => null,
};

export default SimpleButton;
