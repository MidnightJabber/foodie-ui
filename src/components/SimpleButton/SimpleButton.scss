@import '~assets/scss/colors';

.simplebutton {
    display: flex;
    color: $colors-green;
    height: auto;
    border: 2px solid $colors-green;
    text-decoration: none;
    padding: 10px 30px;
    display: inline-block;
    border-radius: 100px;
    transition: all .2s;
    position: relative;
    background: white;

    width: auto;
    font-size: 18px;
    font-weight: 550;
    justify-content: center;
    align-content: center;
    text-align: center;

    &:hover {
        transform: translateY(-3px);
        box-shadow: 0 10px 20px rgba(0, 0, 0, .2);
        cursor: pointer;
    }

    &:active {
        transform: translateY(-1px);
        box-shadow: 0 5px 10px rgba(0, 0, 0, .2);
    }

    &:hover::after {
        transform: scaleX(1.4) scaleY(1.6);
        opacity: 0;
    }

    &:focus {
        outline: 0;
    }

    &:after {
        content: "";
        display: inline-block;
        height: 100%;
        width: 100%;
        border-radius: 100px;
        position: absolute;
        top: 0;
        left: 0;
        z-index: -1;
        transition: all .4s;
    }
}
