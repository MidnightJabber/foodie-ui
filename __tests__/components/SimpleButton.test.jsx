import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SimpleButton from 'components/SimpleButton/SimpleButton';

describe('SimpleButton', () => {
  const title = 'Click';

  it('should render without throwing an error', () => {
    expect(shallow(<SimpleButton title={title} />).find('button.simplebutton').exists()).toBe(true);
  });

  it('should set the props correctly', () => {
    const wrapper = shallow(<SimpleButton title={title} />);
    const { className, children } = wrapper.props();
    expect(className).toBe('simplebutton');
    expect(children).toBe(title);
  });

  it('should render deep correctly', () => {
    const tree = renderer.create(<SimpleButton title={title} />);
    expect(tree).toMatchSnapshot();
  });
});
