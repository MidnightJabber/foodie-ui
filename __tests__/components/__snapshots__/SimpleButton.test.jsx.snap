// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton should render deep correctly 1`] = `
<button
  className="simplebutton"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Click
</button>
`;
